Halt on Ddoc errors & fix the existing ones

wilzbach opened the pull request:
Required for https://github.com/dlang/dmd/pull/7440

andralex commented on posix.mak:
```diff
@@ -147,19 +147,19 @@ endif
 
 $(DOCDIR)/object.html : src/object.d
-	$(DMD) $(DDOCFLAGS) -Df$@ project.ddoc $(DOCFMT) $<
+	$(DMD) $(DDOCFLAGS) -Df$@ project.ddoc $(DOCFMT) $< 2>&1 | grep "Warning:"; test $$? -eq 1
```
I think all you need to set the error level is `grep -v "Warning:"`.

andralex replied:
Also a problem with this pattern is that if dmd fails with no warning the command will succeed.

wilzbach commented:
Okay I removed the explicit grepping for error messages and rebased the PR to just contain the Ddoc fix so that it's absolutely non-controversial and we can move on with this, because once this is merged we can merge the dlang/dmd#7440 at DMD and then `-w` will work as expected with a proper exit code and no grepping is required.
